lib/cli/commands/subscribeorders.ts

```diff
@@ -15,20 +15,51 @@ export const builder = {
 };
 
 export const handler = (argv: Arguments) => {
+  ensureConnection(argv, true);
+};
+
+const ensureConnection = (argv: Arguments, printError?: boolean) => {
+  loadXudClient(argv).getInfo(new xudrpc.GetInfoRequest(), (error: Error | null) => {
+    if (error) {
+      if (printError) console.error(`${error.name}: ${error.message}`);
+      setTimeout(ensureConnection.bind(undefined, argv), 3000);
+    } else {
+      console.log('Successfully connected, subscribing for orders');
+      subscribeOrders(argv);
+    }
+  });
+};
+
+const subscribeOrders = (argv: Arguments) =>  {
   const addedOrdersRequest = new xudrpc.SubscribeAddedOrdersRequest();
   addedOrdersRequest.setExisting(argv.existing);
   const addedOrdersSubscription = loadXudClient(argv).subscribeAddedOrders(addedOrdersRequest);
   addedOrdersSubscription.on('data', (order: xudrpc.Order) => {
     console.log(`Order added: ${JSON.stringify(order.toObject())}`);
   });
 
+  // adding end, close, error events only once,
+  // since they'll be thrown for three of subscriptions in the corresponding cases, catching once is enough.
+  addedOrdersSubscription.on('end', ensureConnection.bind(undefined, argv, true));
+  addedOrdersSubscription.on('close', ensureConnection.bind(undefined, argv, true));
+  addedOrdersSubscription.on('error', (err: Error) => {
+    console.log(`Unexpected error occured: ${JSON.stringify(err)}, retrying to connect`);
+    ensureConnection(argv);
+  });
+
   const removedOrdersSubscription = loadXudClient(argv).subscribeRemovedOrders(new xudrpc.SubscribeRemovedOrdersRequest());
   removedOrdersSubscription.on('data', (orderRemoval: xudrpc.OrderRemoval) => {
     console.log(`Order removed: ${JSON.stringify(orderRemoval.toObject())}`);
   });
 
+  // prevent exiting and do nothing, it's already cached above.
+  removedOrdersSubscription.on('error', () => {});
+
   const swapsSubscription = loadXudClient(argv).subscribeSwaps(new xudrpc.SubscribeSwapsRequest());
   swapsSubscription.on('data', (swapResult: xudrpc.SwapResult) => {
     console.log(`Order swapped: ${JSON.stringify(swapResult.toObject())}`);
   });
+
+  // prevent exiting and do nothing, it's already cached above.
+  swapsSubscription.on('error', () => {});
```

Comment:
why do we need this line (and also line 56)? client will throw error if not subscribing to it?

Reply:
No actually these lines are required in case of streaming errors, if we don't catch them like that they'll stop the reading stream

Reply:
I think if there's an error on the stream, it will end anyway (because the server is ending it, nothing the client can do) and adding a listener here won't change it. If we want to reconnect, I think we'd need to do so by making the rpc calls again, which you're already doing. I'm not sure this "catches" anything, but in case I'm wrong could you explain some more? Is there information on this online anywhere or did you run into needing this in testing?